src/com/dynatrace/loadrunner/Main.java

```diff
@@ -1,59 +1,202 @@
+				+ "	-c: sets C as scripting language used (default)\n"
+				+ "	-js: sets JavaScript as scripting language\n"
+				+ "	-help: prints usage\n");
+	}
+
+	public static void main(String[] args) {
+
+		System.out.println(VERSION + "\n");
+
+		if (args.length == 0) {
+			showUsages();
+			return;
+		}
+
+		CommandLineParser parser = new CommandLineParser(args);
+
+		if (parser.arguments.containsKey("-help") || parser.arguments.containsKey("help")) {
+			showUsages();
+			return;
+		}
+
+		for (Entry<String, String> pair : parser.arguments.entrySet()) {
+			String key = pair.getKey().toLowerCase();
+			if (key.equals("insert")) {
+				getMode(pair.getKey());
+			} else if (key.equals("delete")) {
+				getMode(pair.getKey());
+			} else if (key.equals("lsn")) {
+				getScriptName(pair.getValue());
+			} else if (key.equals("path")) {
+				try {
+					File directory = new File(pair.getValue());
+					searchPath(directory);
+				} catch (NullPointerException e) {
+					System.out.println("Make sure you'd provided correct path to script file \n");
+					return;
+				}
+			} else if (key.equals("-c")) {
+				cEngine = true;
+			} else if (key.equals("-js")) {
+				cEngine = false;
+			} else if (key.equals("body")) {
+				String bodies[] = pair.getValue().split("&");
+				for (String str : bodies) {
+					body.add(new ScriptFile(new File(str)));
+				}
+			} else if (key.equals("header")) {
+				String header[] = pair.getValue().split("&");
+				for (String str : header) {
+					headers.add(new ScriptFile(new File(str)));
+				}
+			} else {
+				showUsages();
+				System.out.println("\nUnknown parameter: " + key);
+				return;
+			}
+		}
+		if (scriptName.isEmpty() && parser.containsKey("path")) {
+			getScriptNameFromPath(parser.arguments.get("path"));
+		}
+		checkEngine();
+		boolean validated = validateParams();
+		if (validated) {
+			LoadRunnerConverter converter = new LoadRunnerConverter();
+			converter.configureConverter(mode, headers, body, scriptName, cEngine);
+			converter.convertFiles();
+			System.out.println("conversion complete");
+		} else {
+			System.out.println("ERROR, conversion failed");
+		}
+	}
+
+	private static void getScriptNameFromPath(String path) {
+		File directory = new File(path);
+		for (File file : directory.listFiles()) {
+			if (getFileExtension(file).equals("usr")) {
+				scriptName = file.getName().replaceFirst("[.][^.]+$", "");
+			}
+		}
+	}
+
+	private static boolean validateParams() {
+		if (headers.isEmpty()) {
+			System.out.println("Headers not found");
+			return false;
+		}
+		if (body.isEmpty()) {
+			System.out.println("Bodies not found");
+			return false;
+		}
+		if (mode == null) {
+			System.out.println("Provide insert/delete parameter");
+			return false;
+		}
+		return true;
+	}
+
+	private static void searchPath(final File folder) {
+		for (final File fileEntry : folder.listFiles()) {
```

Comment:
keep the headerFiles and bodyFiles locally, return them using:
`return new Tuple<>(headerFiles, bodyFiles);`